modules/mono/editor/GodotTools/GodotTools/Ides/Rider/RiderPathLocator.cs

```diff
@@ -189,134 +297,187 @@ internal static ProductInfo GetBuildVersion(string path)
+        }
+
+        internal Version GetBuildNumber(string path)
+        {
+            var buildTxtFileInfo = new FileInfo(Path.Combine(path, GetRelativePathToBuildTxt()));
+            return GetBuildNumberWithBuildTxt(buildTxtFileInfo) ?? GetBuildNumberFromInput(path);
         }
 
-        internal static Version GetBuildNumber(string path)
+        private Version GetBuildNumberWithBuildTxt(FileInfo file)
         {
-            var file = new FileInfo(Path.Combine(path, GetRelativePathToBuildTxt()));
             if (!file.Exists)
                 return null;
             var text = File.ReadAllText(file.FullName);
-            if (text.Length <= 3)
+            var index = text.IndexOf("-", StringComparison.Ordinal) + 1; // RD-191.7141.355
+            if (index <= 0)
+                return null;
+
+            var versionText = text.Substring(index);
+            return GetBuildNumberFromInput(versionText);
+        }
+
+        [CanBeNull]
+        private Version GetBuildNumberFromInput(string input)
+        {
+            if (string.IsNullOrEmpty(input))
                 return null;
 
-            var versionText = text.Substring(3);
-            return Version.TryParse(versionText, out var v) ? v : null;
+            var match = Regex.Match(input, @"(?<major>\d+)\.(?<minor>\d+)\.(?<build>\d+)");
+            var groups = match.Groups;
+            Version version = null;
+            if (match.Success)
+            {
+                version = new Version($"{groups["major"].Value}.{groups["minor"].Value}.{groups["build"].Value}");
+            }
+
+            return version;
         }
 
-        internal static bool IsToolbox(string path)
+        public bool GetIsToolbox(string path)
         {
-            return path.StartsWith(GetToolboxBaseDir());
+            return Path.GetFullPath(path).StartsWith(Path.GetFullPath(GetAppsRootPathInToolbox()));
         }
 
-        private static string GetRelativePathToBuildTxt()
+        private string GetRelativePathToBuildTxt()
         {
-            if (OS.IsWindows || OS.IsUnixLike)
-                return "../../build.txt";
-            if (OS.IsMacOS)
-                return "Contents/Resources/build.txt";
-            throw new InvalidOperationException("Unknown OS.");
+            switch (RiderLocatorEnvironment.operatingSystemFamily)
+            {
+                case OperatingSystemFamily.Windows:
+                case OperatingSystemFamily.Linux:
+                    return "../../build.txt";
+                case OperatingSystemFamily.MacOSX:
+                    return "Contents/Resources/build.txt";
+            }
+
+            throw new Exception("Unknown OS");
         }
 
-        [SupportedOSPlatform("windows")]
-        private static void CollectPathsFromRegistry(string registryKey, List<string> installPaths)
+        private void CollectPathsFromRegistry(string registryKey, List<string> installPaths)
```

Comment:
This method shouldn't remove the supported OS platform attribute. Using the attribute ensures that we don't accidentally call this method from an unsupported OS, in this case APIs that try to access Windows' registry will throw an exception in a non-Windows OS.